contracts/neutron_interchain_queries/src/contract.rs

```diff
@@ -241,6 +284,28 @@ fn query_recipient_txs(deps: Deps<NeutronQuery>, recipient: String) -> NeutronRe
     Ok(to_binary(&GetRecipientTxsResponse { transfers: txs })?)
 }
 
+pub fn query_cw20_balance(
+    deps: Deps<NeutronQuery>,
+    _env: Env,
+    registered_query_id: u64,
+) -> NeutronResult<Cw20BalanceResponse> {
+    let registered_query = get_registered_query(deps, registered_query_id)?;
+
+    check_query_type(registered_query.registered_query.query_type, QueryType::KV)?;
+
+    let mut response = get_interchain_query_result(deps, registered_query_id)?;
+
+    // we expect exactly one storage value
+    let result = response
+        .result
+        .kv_results
+        .pop()
+        .ok_or_else(|| StdError::generic_err("empty query result"))?;
+
+    let balance: Uint128 = from_binary(&result.value)?;
```

Comment:
Why is it not implemented using `KVReconstruct` trait like other out interchain queries responses?

Reply:
Again, layout of this particular response is contract-dependent, hence I decided to avoid using KVReconstruct.

Reply:
Alright, I have implemented KVReconstruct for Uint128, that's better than nothing =)